Library/ViewModels/AddNewCardViewModel.swift

```diff
@@ -288,8 +300,39 @@ public final class AddNewCardViewModel: AddNewCardViewModelType, AddNewCardViewM
   public let rememberThisCardToggleViewControllerIsOn: Signal<Bool, Never>
   public let saveButtonIsEnabled: Signal<Bool, Never>
   public let setStripePublishableKey: Signal<String, Never>
+  public let unsupportedCardBrandError: Signal<String, Never>
   public let zipcodeTextFieldBecomeFirstResponder: Signal<Void, Never>
 
   public var inputs: AddNewCardViewModelInputs { return self }
   public var outputs: AddNewCardViewModelOutputs { return self }
 }
+
+private func cardBrandIsSupported(project: Project?, cardNumber: String) -> Bool {
+  guard let project = project else { return true }
+
+  let allOthers = project.location.country != Project.Country.us.countryCode
+  let brand = STPCardValidator.brand(forNumber: cardNumber)
+
+  let allCardBrands: [STPCardBrand] = [
+    .amex,
+    .dinersClub,
+    .discover,
+    .JCB,
+    .masterCard,
+    .unionPay,
+    .visa
+  ]
+
+  let unsupportedCardBrands: [STPCardBrand] = [
+    .dinersClub,
+    .discover,
+    .JCB
+  ]
```

Comment:
I think there was a small misunderstanding as to the requirements of this ticket. The supported card types should actually be coming from `project.availableCardTypes`, not hard-coded on the client anymore. You should be able to transform `project.availableCardTypes` from type `[String]` to type `[GraphUserCreditCard.CreditCardType]`, and from there just check if the card brand is contained in the array of supported cards. Something like this:

```
private func cardBrandIsSupported(project: Project, cardBrand: GraphUserCreditCard.CreditCardType) -> Bool {
  guard let availableCardTypes = project.availableCardTypes else {
    return true
  }

  let availableCreditCardTypes = availableCardTypes
    .compactMap { GraphUserCreditCard.CreditCardType(rawValue :$0) }

  return availableCreditCardTypes.contains(cardBrand)
}
```